### DISCUSSION TITLE
How to use "unicode"?

### DISCUSSION THREAD

**Discussion by joewinbpaul** [cssbattle/forum#329] [Q&A] [answered]
I am a player in _CSS battle_ and I am always surprised by how the experts use Unicode to replicate the targets _perfectly_.

Can anyone tell me how to use Unicode in CSS?

I want to know...

**Comment by aarthvaen** [comment, marked as answer]
To get started on Unicode, you will need this CSSBattle  [previewer](https://tc70f3.csb.app/). It's no different from the original editor, however, it helps you get accurate renders of your Unicode character(s).

Because people use different browsers when playing CSSBattle, these HTML entities are rendered differently on each browser, hence why you'll not get 100% match on your solution, although the Unicode character is "placed" where needed.

The previewer linked above also includes a collection of characters used by most Unicode Top Solutions. The magic of recreating the image with Unicode and also little CSS code is something you will need to figure out on your own. Some Open Targets are good as to understand the most efficient formatting.

**Reply by joewinbpaul** [reply]
THX
But I also wanted to know exactly HOW we place them or shape them a little like in Target #91 where the full stop-like Unicode was changed into lines.

I know that the "font" property helps, but I don't know how to use it and couldn't find any tutorials.

Pls answer this

**Reply by SelenIT** [reply]
The open solution of [#91](https://cssbattle.dev/play/91) is not a real unicode solution, those full stop characters just make the `<p>` elements nonempty, so the line height (the part of the `font` value after the slash, i.e. `5vh` or 15px) serves as a height placeholder. The dots themselves are virtually invisible because of really small font (font size of `<p>`s is 9% of the font size of `body` which is 9% of the font size of `html` which is 9% of default 16px, i.e. 16 * 0.09³ ≈ 0.01px). And the background of those 15px-tall paragraphs becomes green because the gradient in them has different color stops (if I understand the specs correctly, the yellow area shouldn't shrink to the green area boundary, so it seems to be technically a rendering bug, but it's quite fortunate for this target:)

**Reply by joewinbpaul** [reply]
> The open solution of [#91](https://cssbattle.dev/play/91) is not a real unicode solution, those full stop characters just make the `<p>` elements nonempty, so the line height (the part of the `font` value after the slash, i.e. `5vh` or 15px) serves as a height placeholder. The dots themselves are virtually invisible because of really small font (font size of `<p>`s is 9% of the font size of `body` which is 9% of the font size of `html` which is 9% of default 16px, i.e. 16 * 0.09³ ≈ 0.01px). And the background of those 15px-tall paragraphs becomes green because the gradient in them has different color stops (if I understand the specs correctly, the yellow area shouldn't shrink to the green area boundary, so it seems to be technically a rendering bug, but it's quite fortunate for this target:)

So what u mean is that the full stop characters were just helping the paragraph tags and not being converted to lines, right?

**Reply by SelenIT** [reply]
basically yes, any text inside paragraphs + line-height of 15px is a shorter equivalent to giving them `height:15px`

**Reply by joewinbpaul** [reply]
> basically yes, any text inside paragraphs + line-height of 15px is a shorter equivalent to giving them `height:15px`

...... I am getting ideas for getting higher scores now....

**Reply by joewinbpaul** [reply]
> ...... I am getting ideas for getting higher scores now....

But at the same time... I'm not

**Reply by eguiluz** [reply]
Could you explain the use of the last quote used in the font attribute?

font:9%/5vh'

Thanks!

**Reply by aarthvaen** [reply]
That \` quote is basically the spot reserved for a font-family. It is not needed to place a single quote, also a single letter (a-z) is allowed, but placing a single quote is just 1 char shorter because you need a space in front of the letter notation. In this case, `font:9%/5vh'` will be shorter than typing `font:9%/5vh a`.

**Reply by eguiluz** [reply]
Wow, I find it amazing, I've been testing for days to see if I could figure out the hack myself but I've been unable to find a logic.
Thanks @almostvanished

**Comment by joewinbpaul** [comment]
Btw is it possible to "shape"/_change the structure_ of an unicode? <p> I mean like is it possible to make a rectangle ( ▅ ) out of a square Unicode character ( ▖)?

**Reply by joewinbpaul** [reply]
> Btw is it possible to "shape"/_change the structure_ of an unicode?
> 
> I mean like is it possible to make a rectangle ( ▅ ) out of a square Unicode character ( ▖)?

Or stretch a circle unicode ( ⬤ ) to make it an oval?

**Reply by SelenIT** [reply]
It's possible, e.g. by using non-uniform `scale` (like `scale:10 5` to stretch the character horizontally twice as much as vertically). But at the same time, there are chances that other characters may fit better without such adjustment. For example, the ["Block Elements"](https://en.wikipedia.org/wiki/Block_Elements) group contains a set of rectangles of different proportions and even checquered patterns as single characters

**Reply by joewinbpaul** [reply]
thx cuz I wanted to break my record in target 34 using unicode 😅😅 <p> And now I did it with 114 chars

**Reply by joewinbpaul** [reply]
And not only that target but some others as well <p> Now I'm starting to like Unicode
